plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -735,3 +747,42 @@ func haveSame(a1, a2 []string) bool {
 	}
 	return false
 }
+
+type NodeStatus struct {
+	info NodeInfo
+}
+
+func GeneralPredicates(info NodeInfo) algorithm.FitPredicate {
+	node := &NodeStatus{
+		info: info,
+	}
+	return node.SchedulerGeneralPredicates
+}
+
+func (n *NodeStatus) SchedulerGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+	node, err := n.info.GetNodeInfo(nodeName)
+	if err != nil {
+		return false, err
+	}
+	return RunGeneralPredicates(pod, nodeName, nodeInfo, node)
+}
+
+func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo, node *api.Node) (bool, error) {
+	fit, err := PodFitsResource2(pod, nodeName, nodeInfo, node)
+	if !fit {
+		return fit, err
+	}
+	fit, err = PodFitsHost(pod, nodeName, nodeInfo)
+	if !fit {
+		return fit, err
+	}
+	fit, err = PodFitsHostPorts(pod, nodeName, nodeInfo)
+	if !fit {
+		return fit, err
+	}
+	// TODO: decide whether to keep this as part of GeneralPredicates()
```

Comment:
@dchen1107 @bgrant0607 : If the commented-out code is added to this function, then Kubelet will start checking nodeSelector (and node affinity) as part of admission. Personally, I think we should make the change now and announce the change, because although some people may be using Kubelet directly (that's the only case where anyone could notice the change, since scheduler already checks this predicates), if we wait for the next release even more people may be using Kubelet directly. WDYT?